src/reducer.js

```diff
@@ -75,6 +88,9 @@ export default function reducer (state, event) {
   if (event.type === 'visit') {
     // Increment the number of user visits.
     user = incrementVisits(user)
+  } else if (event.type === 'click') {
+    // Add promo to the list of engagements.
+    user = trackEngagements(event.promo, ts, user)
```

Comment:
Here we track the engagement for any `click` events. This means that the `engagements` in the user state object will be updated in exactly the same as we do for impressions and close events.